File: reference-app/src/main/java/org/smartregister/sample/fp/app/FPSyncConfiguration.java
package org.smartregister.sample.fp.app;

import org.smartregister.SyncConfiguration;
import org.smartregister.SyncFilter;
import org.smartregister.fp.common.library.FPLibrary;
import org.smartregister.repository.AllSharedPreferences;
import org.smartregister.sample.fp.BuildConfig;

import java.util.List;

/**
 * Created by samuelgithengi on 10/19/18.
 */
public class FPSyncConfiguration extends SyncConfiguration {
    @Override
    public int getSyncMaxRetries() {
        return BuildConfig.MAX_SYNC_RETRIES;
    }

    @Override
    public SyncFilter getSyncFilterParam() {
        return SyncFilter.TEAM_ID;
    }

    @Override
    public String getSyncFilterValue() {
        AllSharedPreferences sharedPreferences =
                FPLibrary.getInstance().getContext().userService().getAllSharedPreferences();
        return sharedPreferences.fetchDefaultTeamId(sharedPreferences.fetchRegisteredANM());
    }

    @Override
    public int getUniqueIdSource() {
        return BuildConfig.OPENMRS_UNIQUE_ID_SOURCE;
    }

    @Override
    public int getUniqueIdBatchSize() {
        return BuildConfig.OPENMRS_UNIQUE_ID_BATCH_SIZE;
    }

    @Override
    public int getUniqueIdInitialBatchSize() {
        return BuildConfig.OPENMRS_UNIQUE_ID_INITIAL_BATCH_SIZE;
    }

    @Override
    public boolean isSyncSettings() {
        return BuildConfig.IS_SYNC_SETTINGS;
    }

    @Override
    public SyncFilter getEncryptionParam() {
        return SyncFilter.TEAM;
    }

    @Override
    public boolean updateClientDetailsTable() {
        return true;
    }

    @Override
    public List<String> getSynchronizedLocationTags() {
        return null;
    }

    @Override
    public String getTopAllowedLocationLevel() {
        return null;
    }
}

File: reference-app/src/main/java/org/smartregister/sample/fp/app/FPSyncConfiguration.kt
package org.smartregister.sample.fp.app

import org.smartregister.SyncConfiguration
import org.smartregister.SyncFilter
import org.smartregister.sample.fp.BuildConfig

<<<<<<< HEAD:reference-app/src/main/java/org/smartregister/sample/fp/app/FPConfiguration.kt
class FPConfiguration : SyncConfiguration() {
=======
class FPSyncConfiguration : SyncConfiguration() {
>>>>>>> issue5-create-main-register-page:reference-app/src/main/java/org/smartregister/sample/fp/app/FPSyncConfiguration.kt
    override fun getSyncMaxRetries(): Int {
        return BuildConfig.MAX_SYNC_RETRIES
    }

    override fun getSyncFilterParam(): SyncFilter {
        return SyncFilter.TEAM_ID
    }

    override fun getSyncFilterValue(): String {
//        AncLibrary will design that later
        /*val sharedPreferences: AllSharedPreferences = AncLibrary.getInstance().getContext().userService().getAllSharedPreferences()
        return sharedPreferences.fetchDefaultTeamId(sharedPreferences.fetchRegisteredANM())*/
        return ""
    }

    override fun getUniqueIdSource(): Int {
        return BuildConfig.OPENMRS_UNIQUE_ID_SOURCE
    }

    override fun getUniqueIdBatchSize(): Int {
        return BuildConfig.OPENMRS_UNIQUE_ID_BATCH_SIZE
    }

    override fun getSynchronizedLocationTags(): MutableList<String> {
        return ArrayList()
    }

    override fun getUniqueIdInitialBatchSize(): Int {
        return BuildConfig.OPENMRS_UNIQUE_ID_INITIAL_BATCH_SIZE
    }

    override fun isSyncSettings(): Boolean {
        return BuildConfig.IS_SYNC_SETTINGS
    }

    override fun getEncryptionParam(): SyncFilter {
        return SyncFilter.TEAM
    }

    override fun getTopAllowedLocationLevel(): String {
        return ""
    }

    override fun updateClientDetailsTable(): Boolean {
        return true
    }

    override fun getSynchronizedLocationTags(): List<String?>? {
        return null
    }

    override fun getTopAllowedLocationLevel(): String? {
        return null
    }
}
